test/dsl/dsl_options.jl

```diff
@@ -152,6 +153,68 @@ let
     @test issetequal(parameters(rn5), @parameters k B X)
 end
 
+# Test whether user-defined functions are properly expanded in equations. 
+let
+    f(A, t) = 2*A*t
+
+    # Test user-defined function
+    rn = @reaction_network begin
+        @equations D(A) ~ f(A, t)
+    end
+    @test length(equations(rn)) == 1
+    @test equations(rn)[1] isa Equation
+    @species A(t)
+    @test isequal(equations(rn)[1], D(A) ~ 2*A*t)
+
+
+    hill_unregistered(A, v, K, n) = v*(A^n) / (A^n + K^n)
+    rn2 = @reaction_network begin
+        @parameters v K n
+        @equations D(A) ~ hill_unregistered(A, v, K, n)
+    end
+    @test length(equations(rn2)) == 1
+    @test equations(rn2)[1] isa Equation
+    @parameters v K n
+    @test isequal(equations(rn2)[1], D(A) ~ v*(A^n) / (A^n + K^n))
+
+    @register_symbolic hill2(A, v, K, n)
+    # Test Catalyst function
+    rn2r = @reaction_network begin
+        @parameters v K n
+        @equations D(A) ~ hill2(A, v, K, n)
+    end
+    @test length(equations(rn2r)) == 1
+    @test equations(rn2r)[1] isa Equation
+    @parameters v K n
+    @test isequal(equations(rn2r)[1], D(A) ~ hill2(A, v, K, n))
+
```

Comment:
Is this what you had in mind for the registered/unregistered function test @isaacsas?

Reply:
This is what I had in mind for the two ways of specifying external functions, yes. 

I'd suggest making two versions of the system with the same name (one constructed via the symbolic interface), then you can test if they are equal. That would establish that the DSL interface is generating the same system as the symbolic interface.